apollo-router/src/plugins/telemetry/config_new/apollo/instruments.rs

```diff
@@ -0,0 +1,184 @@
+use crate::services::subgraph;
+
+pub(crate) struct ApolloSubgraphInstruments {
+    pub(crate) apollo_router_operations_fetch_duration: Option<
+        CustomHistogram<
+            subgraph::Request,
+            subgraph::Response,
+            (),
+            SubgraphAttributes,
+            SubgraphSelector,
+        >,
+    >,
+}
+
+impl ApolloSubgraphInstruments {
+    pub(crate) fn new(
+        static_instruments: Arc<HashMap<String, StaticInstrument>>,
+        apollo_config: Config,
+    ) -> Self {
+        let selectors = Extendable {
+            attributes: SubgraphAttributes::builder()
+                .subgraph_name(StandardAttribute::Bool(true))
+                .build(),
+            custom: HashMap::from([
+                (
+                    "client.name".to_string(),
+                    SubgraphSelector::ResponseContext {
+                        response_context: CLIENT_NAME.to_string(),
+                        redact: None,
+                        default: None,
+                    },
+                ),
+                (
+                    "client.version".to_string(),
+                    SubgraphSelector::ResponseContext {
+                        response_context: CLIENT_VERSION.to_string(),
+                        redact: None,
+                        default: None,
+                    },
+                ),
+                (
+                    "graphql.operation.name".to_string(),
+                    SubgraphSelector::SupergraphOperationName {
+                        supergraph_operation_name: OperationName::String,
+                        redact: None,
+                        default: None,
+                    },
+                ),
+                (
+                    "graphql.operation.type".to_string(),
+                    SubgraphSelector::SupergraphOperationKind {
+                        supergraph_operation_kind: OperationKind::String,
+                    },
+                ),
+                (
+                    "operation.id".to_string(),
+                    SubgraphSelector::ResponseContext {
+                        response_context: APOLLO_OPERATION_ID.to_string(),
+                        redact: None,
+                        default: None,
+                    },
+                ),
+                (
+                    "has.errors".to_string(),
+                    SubgraphSelector::OnGraphQLError {
+                        subgraph_on_graphql_error: true,
+                    },
+                ),
+            ]),
+        };
+
+        let apollo_router_operations_fetch_duration =
+            apollo_config.experimental_subgraph_metrics.then(|| {
+                CustomHistogram {
+                    inner: Mutex::new(
+                        CustomHistogramInner {
+                            increment: Increment::Duration(Instant::now()),
+                            condition: Condition::True,
+                            attributes: Vec::with_capacity(7),
```

Comment:
Must this number be kept in sync with something? Can we derive it instead of hardcoding it?